modules/table/src/main/java/org/apache/ignite/internal/table/KeyValueBinaryViewImpl.java

```diff
@@ -475,8 +476,8 @@ private Tuple unmarshalValue(BinaryRow row) {
      * @param rows Binary rows.
      * @return Key-value pairs of tuples.
      */
-    public Map<Tuple, Tuple> unmarshalValue(Collection<BinaryRow> rows) {
-        Map<Tuple, Tuple> pairs = new HashMap<>();
+    private Map<Tuple, Tuple> unmarshalValue(Collection<BinaryRow> rows) {
+        Map<Tuple, Tuple> pairs = new HashMap<>(rows.size());
```

Comment:
Seems we still obtain reallocations here, check [1]

[1] https://github.com/apache/ignite/blob/master/modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java#L10140